File: BOL/Order.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BOL
{
    public class Order
    {
        //books, quantity
        public IList<Dictionary<Book, int>> OrderedItems { get; set;}
        public string ReceiverName { get; set; }
        public Address ReceiverAddr { get; set; }
        public string ReceiverContactNo { get; set; }
        public bool OrderShipped { get; set; }
        public string OrderTransactonId { get; set; }

        public Order(string ReceiverName, Address ReceiverAddr, string ReceiverContactNo, IList<Dictionary<Book, int>> OrderedItems, bool OrderShipped, string OrderTransactonId)
        {
            this.ReceiverAddr = ReceiverAddr;
            this.ReceiverContactNo = ReceiverContactNo;
            this.ReceiverName = ReceiverName;
            this.OrderedItems = OrderedItems;
            this.OrderTransactonId = OrderTransactonId;
        }
    }
}

File: FrontEnd/src/app/shared/models/order.ts
import { BookCount } from './book';
import { Address } from './Address';

export interface Order {
    OrderId?: string;
    OrderedItems: BookCount[];
    ReceiverName: string;
    ReceiverAddr: Address;
    ReceiverContactNo: string;
    OrderShipped: boolean;
    OrderTransactionId: string;
    OrderPlaceTime?: string;
}

